### ISSUE TITLE
[Enhancement] Asset Continuity

### ISSUE THREAD

**Issue by Bruhnanza** [wisp-forest/accessories#177]
Hello, I noticed while working on a resource pack to retexture Accessories to be compatible with another pack that Accessories has a few quirks related to its assets.
- The primary GUI button uses the `misc.png` texture which is both shaded so the asset color cannot be accurately lightened, and is the same asset used in the Original GUI groups as that is also the folder is under.
- The cosmetic button in the Experimental GUI has a static gradient that uses `cosmetic.png` as an outline meaning no color change there will be displayed.
- The back button will warp to never cleanly display a smaller texture regardless of the image's dimensions.
- The Experimental GUI background does not have a texture (that I can find) unlike the Original GUI.
![1](https://github.com/user-attachments/assets/0a0e1127-cf45-4b49-ad17-f43715b2a754)

These are some of the primary examples, but there may be more. I have a few ideas I hope you may consider implementing that I believe would be a better approach:

1. Reference assets as much as possible (or add an option to?). For the primary and cosmetic buttons, they are both currently unable to be colored correctly with the way things are. If possible add a separate texture for the primary and simply use a rainbow filled cosmetic symbol as assets. An example I feel would be unnecessary to change is the actual cosmetic slots which have a RGB outline which isn't warranting of an animated file nor does it impede the slot assets themselves.
2. The back button would be rendered as the asset without shrinking and use an alternative texture if needed for another GUI if it would shrink. For example, the primary button in Creative mode is rendered much smaller than in Survival mode. I believe that due to the size difference in that regard it is appropriate. But for the back button, I believe it simply ends up messing with the texture for no real benefit.
3. The Experimental GUI should have a background somewhere, whether optional or in the container folder. This request is as the pack I follow, [Stormilla](https://modrinth.com/resourcepack/stormilla), adds shaded backgrounds to GUI where the Experimental one is not capable of that. Below are what I did to the Original GUI with an option to have a divider or not.
![asdasdasd](https://github.com/user-attachments/assets/b5ac9451-ad1b-42cc-8ccf-b9d32c52491d)

In conclusion, I would appreciate it if future Accessories iterations would be more faithful to their assets or allow asset changes to not be restricted by the mod. I hope this all made sense, if you need any additional elaboration I would be happy to clarify. Thank you for your work, it is a wonderful mod and I appreciate all the betas and work that consistently is put into it.

TL;DR plz more consolidated and effective asset usage

**Comment by Dragon-Seeker** [comment]
This will take some time to respond to these issues as its quite alot to take in but I will be making a full response about this. I do plan to adjust things but I want to illustrate the screen is very dynamic using various methods to reorder GUI elements so for instance there is no background image as a owo ui panel background is used which can be changed to match the given screens colors.

**Comment by Bruhnanza** [comment]
I understand and will await your full response. To make things more clear, would it be possible to keep the Original GUI as it is currently with more static elements, then allowing the Experimental GUI to have the more dynamic elements? To better understand where I am coming from, I am retexturing modded assets to be in line with the resource pack I use. Since I would be asking for whole PNGs available to edit I understand that is not the direction you may be taking.

To narrow things that way it is actually solvable for you, I would ask for these 2 changes:
1. Have the Accessories menu button use its own asset separate from the groups icon, that would display true to the image if possible
2. Fix the scaling with the back arrow button inside of the Accessories menu

Just these two would be enough as it would allow me to change all the visual elements of the Original GUI. Everything else would be your discretion going forward, I only wanted to provide my perspective as a user of the mod and a resource pack creator with not a lot of understanding for the mod's backend. I hope this is more understandable, I apologize for the lengthy opening.

**Comment by Dragon-Seeker** [comment]
@Bruhnanza 

An update will use background images instead of dynamic surfaces for elements that are not massively dynamic. However, the sidebar of all accessories is quite scalable, so it will stay a dynamic surface that is nine patch generated. 

All of the assets used to build the GUI's dynamic textures will now be found within accessories, so the buttons and surfaces can be adjusted to match the color.

The back button has been adjusted to allow for smaller icons, but it's done the way it is to keep the back button smaller. I might adjust such in the future if it becomes an issue still.

Please look though 9c233d4b81f4293cdcf2ac83dab484eeae3bc847 to see the changes.

**Comment by Dragon-Seeker** [comment]
An update has been pushed for 1.21.1 and 1.21.4 that attempts to alleviate most issues found here I believe

**Comment by Bruhnanza** [comment]
Awesome! I'll check it out, thank you so much! I'll reopen if there is anything else pops up
